subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java

```diff
@@ -16,12 +16,24 @@
 
 package org.gradle.internal.util;
 
-import static java.lang.String.format;
+import javax.annotation.Nullable;
+import java.math.BigDecimal;
+import java.math.MathContext;
+
+import static java.math.RoundingMode.FLOOR;
 
 /**
  * Utility methods for working with numbers
  */
 public class NumberUtil {
+    public static final int BASE_LOG2 = 10;
+    public static final int KIB_BASE = 1 << BASE_LOG2;
+    private static final int FRACTIONAL_DIGIT_COUNT = 1;
+    private static final MathContext MC = new MathContext(String.valueOf(KIB_BASE).length() + FRACTIONAL_DIGIT_COUNT, FLOOR);
+
+    private static final String[] UNITS = new String[]{" B", " KiB", " MiB", " GiB", " TiB", " PiB", " EiB"};
+    private static final String[] ORDINAL_SUFFIXES = new String[]{"th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th"};
```

Comment:
I always feel a bit uneasy when I see arrays used as constants. Can we use `ImmutableLists` instead?

Reply:
Done